apps/ops/ws.py

```diff
@@ -71,8 +75,10 @@ async def receive_json(self, content, **kwargs):
         user = self.scope['user']
         user_role_ids = await self.get_current_user_role_ids(user)
         has_admin_auditor_role = bool(admin_auditor_role_ids & user_role_ids)
-
-        if not has_admin_auditor_role and task.name in self.user_tasks and task.creator != user:
+        has_view_joblog_perms = await self.has_perms(user, ['audits.view_joblog'])
+        # (有管理员或审计员角色) 或者 (任务是用户自己创建的 或者 有查看任务日志权限), 其他情况没有权限
+        if not (has_admin_auditor_role or (
+                task.name in self.user_tasks and (task.creator == user or has_view_joblog_perms))):
             await self.send_json({'message': 'No permission', 'task': task_id})
             return
 
```

Comment:
The code seems to be mostly fine with minor adjustments such as adjusting indentation and spacing conventions.

Some issues:

1. In the `receive_json` function at line 83, it checks for `self.scope['user']`, but there is no scope object available since async operations are being run in background threads that don't communicate through Django's request context.

Optimization Suggestions:

1. The check for admin rights should only happen once within receive_json(), rather than before it starts executing. This will reduce unnecessary computation on first startup of async servers.
   
In summary:
There are no major deviations from standard practices. Minor improvements could include ensuring the `scope` variable is present when checking permissions, potentially reducing initial startup times during background thread processes.

For further optimizations, please consider these recommendations. If you have additional questions or require detailed explanations about specific sections of code I've reviewed, feel free to ask!